frame/system/src/lib.rs

```diff
@@ -357,6 +357,55 @@ impl From<sp_version::RuntimeVersion> for LastRuntimeUpgradeInfo {
 	}
 }
 
+/// An object to track the currently used extrinsic weight in a block.
+#[derive(Clone, Eq, PartialEq, Default, RuntimeDebug, Encode, Decode)]
+pub struct ExtrinsicsWeight {
+	normal: Weight,
+	operational: Weight,
+}
+
+impl ExtrinsicsWeight {
+	pub fn total(&self) -> Weight {
+		self.normal.saturating_add(self.operational)
+	}
+
+	pub fn add(&mut self, weight: Weight, class: DispatchClass) {
+		match class {
+			DispatchClass::Operational => self.operational = self.operational.saturating_add(weight),
+			_ => self.normal = self.normal.saturating_add(weight),
```

Comment:
better to use the explicit `DispatchClass::Normal` here so that if ever another item is added to the `DispatchClass` enum, we throw an error unless we change this code to handle it.